packages/common/src/add-name-to-message.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ Adds first and last name when First Name and Last Name fields lose focus if name shortcodes aren't present
+*/
+
+import { ENGrid } from ".";
+
+export class addNameToMessage {
+    constructor() {
+        if(!this.shouldRun()) {
+            // Don't run the script if the page isn't email to target
+            return;
+        }
+
+        this.replaceNameShortcode("#en__field_supporter_firstName", "#en__field_supporter_lastName");
+    }
+    
+    shouldRun() {
+        return ENGrid.getPageType() === "EMAILTOTARGET";
+    }
+
+    replaceNameShortcode(fName: string, lName: string) {
+        const firstName: HTMLInputElement | null = document.querySelector(fName);
+        const lastName: HTMLInputElement | null = document.querySelector(lName);
+        let message: HTMLTextAreaElement | null = document.querySelector('[name="contact.message"]');
+        let addedFirstName: boolean = false;
+        let addedLastName: boolean = false;
+        
+        if(ENGrid.getPageType() == "EMAILTOTARGET" && message) {
```

Comment:
The `ENGrid.getPageType() == "EMAILTOTARGET"` verification was already made on the `shouldRun` function.

Reply:
I've removed the repeating page type check.